atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/timestamp/CleanTransactionRange.java

```diff
@@ -84,17 +82,10 @@ protected int executeTimestampCommand(AtlasDbServices services) {
             printer.info("Found and cleaning possibly inconsistent transaction: [start={}, commit={}]",
                     SafeArg.of("startTs", startTs), SafeArg.of("commitTs", commitTs));
 
-            Cell key = Cell.create(rowName, TransactionConstants.COMMIT_TS_COLUMN);
-            toDelete.put(key, value.getTimestamp());  //value.getTimestamp() should always be 0L
-        }
-
-        if (!toDelete.isEmpty()) {
-            kvs.delete(TransactionConstants.TRANSACTION_TABLE, toDelete);
-            printer.info("Delete completed.");
-        } else {
-            printer.info("Found no transactions after the given timestamp to delete.");
+            transactionService.putUnlessExists(startTs, TransactionConstants.FAILED_COMMIT_TS);
```

Comment:
Hmm, this is going to be super slow now since we're not batching anything (and putUnlessExists is also slow). It might require some hackiness to put the 0 timestamp (though this is probably a constant somewhere), but could we modify this to actually use KVS.put or multiPut?

Reply:
Yep, it's a hardcoded constant in the kvs.delete code.

Reply:
-1 to using pue here, this cli is already very slow.

Reply:
The transactionservice takes care of the 0 timestamp. I am looking into batching the writes, note that the slowness was a concern in the PR description at the time of writing this up..